sbi/user_input/user_input_checks.py

```diff
@@ -509,32 +545,32 @@ def prepare_sbi_problem(
         simulator = process_simulator(user_simulator, prior, prior_returns_numpy)
 
         # Check data, returns data with leading batch dimension.
-        x_o, _ = process_x_o(user_x_o, simulator, prior)
+        x_shape, _ = process_x_shape(simulator, prior, user_x_shape)
 
         # Consistency check after making ready for SBI.
-        check_sbi_problem(simulator, prior, x_o)
+        check_sbi_problem(simulator, prior, x_shape)
 
-        return simulator, prior, x_o
+        return simulator, prior, x_shape
 
 
-def check_sbi_problem(simulator: Callable, prior, x_o: Tensor):
+def check_sbi_problem(simulator: Callable, prior, x_shape: torch.Size):
```

Comment:
if in `prepare_sbi_problem` the prior return type is `Distribution` then it can be `Distribution` here too.